projects/fsastorefrontlib/src/cms-components/my-account/claim/change-claim/change-claim-navigation.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { RoutingService } from '@spartacus/core';
+import { FileService, FormDataService } from '@spartacus/dynamicforms';
+import { ClaimService } from '../../../../core/my-account/facade/claim.service';
+import { combineLatest, Subscription } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-fs-change-claim-navigation',
+  templateUrl: './change-claim-navigation.html',
+})
+export class ChangeClaimNavigationComponent implements OnInit, OnDestroy {
+  constructor(
+    protected routingService: RoutingService,
+    protected claimService: ClaimService,
+    protected fileService: FileService,
+    protected formDataService: FormDataService
+  ) {}
+
+  claim$;
+  subscription = new Subscription();
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.routingService
+        .getRouterState()
+        .pipe(
+          tap(routingData => {
+            const claimId = routingData.state.params.claimId;
+            if (claimId) {
+              this.claimService.loadClaimById(claimId);
+            }
+          })
+        )
+        .subscribe()
+    );
+    this.claim$ = this.claimService.getCurrentClaim();
```

Comment:
Maybe you could move this initialization above when you declare variable `claim$`?

Reply:
We have to get the current claim after loading and outside of subscription

Reply:
Yes, I know. I did the same in the `QuoteComparisonComponent` and it works. `getCurrentClaim ` will be available or exist only when you call `loadClaimById`, so it doesn't matter if you set `claim$` before or after calling `loadClaimById`. This is just a suggestion to put it into one line of code instead of two :)

Reply:
Ok, moved :)